prow/git/v2/interactor.go

```diff
@@ -220,16 +220,17 @@ func (i *interactor) BranchExists(branch string) bool {
 	return err == nil
 }
 
-func (i *interactor) CommitExists(sha string) (bool, error) {
-	i.logger.WithField("SHA", sha).Info("Checking if SHA exists")
-	_, err := i.executor.Run("branch", "--contains", sha)
-	if err != nil && strings.Contains(err.Error(), "no such commit") {
+func (i *interactor) ObjectExists(sha string) (bool, error) {
+	i.logger.WithField("SHA", sha).Info("Checking if Git object exists")
+	output, err := i.executor.Run("cat-file", "-e", sha)
+	// If the object does not exist, cat-file will exit with a non-zero exit
+	// code.  This will make err non-nil. However this is a known behavior, so
+	// we just log it.
+	if err != nil {
+		i.logger.WithError(err).WithField("SHA", sha).Debugf("error from 'git cat-file -e': %s", string(output))
 		return false, nil
```

Comment:
Hmm, I guess we don't need to return an error tuple any more. Will update.

Reply:
Actually, I think I need to keep this around because the v1 adapter for it needs to return an error saying "ObjectExists is not supported in v1".